### PR TITLE
fix: implement aria-hidden and change aside to nav

### PR DESCRIPTION
**Pull request by dithizac000**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [x] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [x] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

I am a student from Seattle working in team project with my mentors. This ticket was part of my introduction to open resources and using node.js, react, and github. 

## Test Plan

In complementary to ticket #8429, ...
// logo annoucement 

// nav sidebar introduction...  complementary

### Test links
Use screen reader to announce the aria label on logo...
// revert unwanted change 
    // new video 
// resolve conflict
 
 
 
 
 
 
 
 
 
🙏 Please add an exhaustive list of links relevant to this pull request.
⏱ This saves maintainers a lot of time during reviews.

- If you changed anything that's displayed on UI, please add a dogfooding page in website/_dogfooding to help us preview the effect. Those tests are deployed at https://docusaurus.io/tests
- If you changed documentation, please link to the new and updated documentation pages.

After submission, our Netlify bot will post a deploy preview link in comment, in the format of https://deploy-preview-<PR-NUMBER>--docusaurus-2.netlify.app/. Once available, please edit this section with links to the relevant deploy preview pages.

Please don't be afraid to change the main site's configuration as well! You can make use of your new feature on our site so we can preview its effects. We can decide if it should be kept in production before merging it.
-->

Deploy preview: https://deploy-preview-_____--docusaurus-2.netlify.app/

### CODE REVIEW

**Comment by slorber** [comment]
Hi

You marked this PR as a draft so I'm assuming it's not yet ready to review?

Once it is ready for review, please provide more instructions on how I should verify it's working as intended. I'm not an accessibility expert and your instructions are quite unclear to me atm.

File: packages/docusaurus-utils/src/markdownLinks.ts [hunk 1]
```diff
@@ -103,7 +103,7 @@ export function replaceMarkdownLinks<T extends ContentPaths>({
     // This is [Document 1](doc1.md)
     // [doc1]: doc1.md
     const mdRegex =
-      /(?:\]\(|\]:\s*)(?!https?:\/\/|@site\/)<?(?<filename>[^'"\]\s>]+(?:\s[^'"\]\s>]+)*\.mdx?)>?/g;
+      /(?:\]\(|\]:\s*)(?!https?:\/\/|@site\/)(?<filename>[^'")\]\s>]+\.mdx?)/g;
```
Comment (slorber) [inline packages/docusaurus-utils/src/markdownLinks.ts:106]: I don't understand why this PR changes this code and delete snapshots, that looks unrelated to me

**Comment by dithizac000** [comment]
Absolutely!

File: packages/docusaurus-utils/src/markdownLinks.ts (hunk 1 above)
Comment (dithizac000) [reply packages/docusaurus-utils/src/markdownLinks.ts:106]: that was a mistake, about to fix this soon.
